js/switcheo.js

```diff
@@ -0,0 +1,370 @@
+            'show_inactive': 1,
+        }, params));
+        let contracts = await this.fetchContractHash ();
+        let result = [];
+        for (let p = 0; p < markets.length; p++) {
+            let market = markets[p];
+            let id = market['name'];
+            let base = market['name'].split ('_')[0];
+            let quote = market['name'].split ('_')[1];
+            let symbol = base + '/' + quote;
+            let active = (tokens[base]['trading_active'] && tokens[quote]['trading_active']);
+            this.options['contract'] = contracts[quote];
+            let precision = {
+                'amount': market['precision'],
+                'cost': market['precision'],
+                'price': market['precision'],
+            };
+            let limits = {
+                'amount': {
+                    'min': tokens[quote]['minimum_quantity'],
+                },
+                'cost': {
+                    'min': tokens[base]['minimum_quantity'],
+                },
+            };
+            result.push (this.extend ({
+                'id': id,
+                'symbol': symbol,
+                'base': base,
+                'quote': quote,
+                'network': quote,
+                'active': active,
+                'precision': precision,
+                'limits': limits,
+                'info': market,
+            }));
+        }
+        return result;
+    }
+
+    async fetchPairs () {
+        let pairs = await this.publicGetExchangePairs ();
+        return pairs;
+    }
+
+    async fetchOrderBook (symbol, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        let market = this.market (symbol);
+        let request = {
+            'pair': market['id'],
+            'contract': this.options['contract'],
+        };
+        if (limit !== undefined) {
+            request['depth'] = limit;
+        }
+        let response = await this.publicGetOffersBook (this.extend (request, params));
+        return this.parseOrderBook (response, undefined, 'bids', 'asks', 'price', 'quantity');
+    }
+
+    signOrderList (orderParams, privateKey) {
+        let orderDict = {};
+        for (let i = 0; i < orderParams.length; i++) {
+            let signedOrder = this.signMessageHash (orderParams[i]['txn']['sha256'], privateKey);
+            orderDict[orderParams[i]['id']] = '0x' + signedOrder;
+        }
+        return orderDict;
+    }
+
+    signCreateOrder (createOrderDetails, privateKey) {
+        let executeOrder = {
+            'signatures': {
+                'fill_groups': this.signOrderList (createOrderDetails['fill_groups'], privateKey),
+                'fills': {},
+                'makes': this.signOrderList (createOrderDetails['makes'], privateKey),
+            },
+        };
+        return executeOrder;
+    }
+
+    async createOrder (symbol, type, side, amount, price = undefined, useSwitcheoToken = undefined, params = {}) {
```

Comment:
We can't add non-unified arguments to unified method signatures, unfortunately. So, `useSwitcheoToken` has to be passed via overrideable exchange-specific `params`. More about it here: https://github.com/ccxt/ccxt/wiki/Manual#overriding-unified-api-params

Reply:
Removed the `useSwitcheoToken` parameter from the `createOrder` function and ensured that the return object matches the unified message.

This commit should reflect the changes: https://github.com/ccxt/ccxt/pull/4822/commits/578ffcc7c064b4dc35878a14580231531782d72d